pkg/apiserver/domain/service/env.go

```diff
@@ -222,9 +205,7 @@ func (p *envServiceImpl) UpdateEnv(ctx context.Context, name string, req apisv1.
 		return nil, bcode.ErrEnvTargetConflict
 	}
 
-	var targetChanged bool
 	if len(req.Targets) > 0 && !checkEqual(env.Targets, req.Targets) {
```

Comment:
Shall we check if any APP exists(deployed) in this ENV? If this ENV already has APP, reject the update request.
That means user need delete APP before updating ENV's target.

Reply:
I think we do not need to check the app when adding a new target to the environment. For deleting the targets, maybe we should check. What do you think?

Reply:
> I think we do not need to check the app when adding a new target to the environment. For deleting the targets, maybe we should check. What do you think?

Yes, I think
1. if user only update 'description' or 'alias', no need pre-check.
2. if user want to remove some targets from ENV, need pre-check.
3. if user want to add more targets, no need pre-check.

Reply:
@StevenLeiZhang Could review again.